src/debug/logger.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015, 2019 
+ * LAAS-CNRS
+ * Andrea Del Prete, François Bailly 
+ * 
+ * This file is part of dynamic-graph.
+ * See license file.
+ */
+
+#ifndef WIN32
+  #include <sys/time.h>
+#else
+  #include <Windows.h>
+#endif
+
+#include <stdio.h>
+#include <iostream>
+#include <iomanip>      // std::setprecision
+#include <dynamic-graph/logger.h>
+
+namespace dynamicgraph
+{
+
+    using namespace std;
+
+    Logger& getLogger()
+    {
+      static Logger l(0.001, 1.0);
+      return l;
+    }
+
+    Logger::Logger(double timeSample, double streamPrintPeriod)
+      : m_timeSample(timeSample),
+        m_streamPrintPeriod(streamPrintPeriod),
+        m_printCountdown(0.0)
+    {
+#ifdef LOGGER_VERBOSITY_ERROR
```

Comment:
This is will be a problem for binary distribution. The user will lost some flexibility because the flags for the binary will impose the verbosity level. This is a typical case where template programming would be better.